artificer/src/test/java/net/technolords/tools/artificer/reference/JavaVersionManagerTest.java

```diff
@@ -172,13 +174,40 @@ public void testMagicNumberExtractionWithValidClasses(String fileName, String ex
         }
     }
 
+    /** Invalid Classes Data Set to test the Magic Number Extraction.
+    */
+    @DataProvider (name = "dataSetWithInvalidClasses")
+    public Object[][] dataSetWithInvalidClasses() {
+        return new Object[][] {
+                { "abc.class", ArtificerException.class},
+                { "iAmEmpty.class", EOFException.class}
+        };
+    }
     /**
      * Test case 5: Test the extraction of a magic number from a class file as functionality, using the data set
      * containing invalid classes.
      */
-    @Test
-    public void testMagicNumberExtractionWithInvalidClasses() {
-        // TODO: implement, using abc.class and what not...
+    @Test(dataProvider = "dataSetWithInvalidClasses")
+    public void testMagicNumberExtractionWithInvalidClasses(String fileName, Class expectedException) throws ArtificerException, IOException {
+        //TODO: to be reviewed by Mike
+
+        try{
+            // Create a path to the file
+            Path pathToResourceLocation = FileSystems.getDefault().getPath(this.pathToDataFolder.toAbsolutePath() + File.separator + "invalidClass" + File.separator + fileName);
+            LOGGER.debug("The path towards the class file '" + fileName + "' exists: " + Files.exists(pathToResourceLocation));
+
+            // Create a resource reference linking to the file
+            Resource resource = new Resource();
+            resource.setPath(pathToResourceLocation);
+
+            // Find the magic number associated to the resource
+            JavaVersionManager javaVersionManager = new JavaVersionManager(KNOWN_JAVA_VERSIONS_REFERENCE_FILE);
+            String actualVersion = javaVersionManager.getMagicNumber(resource);
+
```

Comment:
Since you expect exceptions, a 'success' is not expected and actually cause the test to fail. This fail is missing here